### PR TITLE
[PriceHistory][UI] ChannelPricingLogEntry implementation

### PR DESCRIPTION
**Pull request by Rafikooo**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13                   |
| Bug fix?        | no                                                      |
| New feature?    |yes                                                       |
| BC breaks?      | no                                                      |
| Deprecations?   | no |
| Related tickets | https://github.com/Sylius/Sylius/pull/14854            |
| License         | MIT                                                          |

image

### CODE REVIEW

File: src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php [hunk 1]
```diff
@@ -203,6 +203,153 @@ public function iDeleteThem(): void
         $this->indexPage->bulkDelete();
     }
 
+    /**
```
Comment (Rafikooo) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php:206]: I noticed that these methods are not sorted with `@when @then`.

File: src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php [hunk 2]
```diff
@@ -203,6 +203,153 @@ public function iDeleteThem(): void
...
+     */
+    public function iWantToModifyAProduct(ProductVariantInterface $productVariant): void
+    {
+        $this->updatePage->open(['id' => $productVariant->getId(), 'productId' => $productVariant->getProduct()->getId()]);
+    }
+
+    /**
+     * @When I choose to show this product in the :channel channel
+     */
+    public function iChooseToShowThisProductInTheChannel(string $channel): void
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php:228]:
```suggestion
     * @When I choose to show this product in the :channelName channel
     */
    public function iChooseToShowThisProductInTheChannel(string $channelName): void
```

File: src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php [hunk 3]
```diff
@@ -203,6 +203,153 @@ public function iDeleteThem(): void
...
+    {
+        $this->generatePage->generate();
+    }
+
+    /**
+     * @When /^I specify that the (\d)(?:st|nd|rd|th) variant is identified by "([^"]+)" code and costs "(?:€|£|\$)([^"]+)" in (("[^"]+") channel)$/
+     */
+    public function iSpecifyThereAreVariantsIdentifiedByCodeWithCost(
+        $nthVariant,
+        $code,
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php:264]: Missing typehints here and in below methods

File: src/Sylius/Bundle/AdminBundle/Resources/translations/messages.en.yml [hunk 4]
```diff
@@ -53,11 +53,15 @@ sylius:
...
             dates_details: This operation is time-consuming! Please consider a 2-10 minutes delay starting from the specified dates. The delay depends on the size of the catalog.
+        channel_pricing_history: 'Channel pricing history'
         channel_pricings: Channel pricings
         product:
             product_not_active_in_channel: The product is not yet activated in this channel.
         original_price_details: Original price - this is the price of the product variant It is displayed as crossed-out in the catalog. It is used as the base for current price calculations. If this value is not defined, Catalog Promotion logic will copy value from Price to Original Price.
         price_details: Price - this is the current price of the product variant displayed in the catalog. It can be modified explicitly by i.e. catalog promotions.
+        price_history: Price history
+        show_history_for_channel_pricing: 'Show history for channel pricing'
+        the_lowest_price_before_the_discount: 'Lowest price before discount'
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Resources/translations/messages.en.yml:64]: Wrong order of keys, and this one should be probably adjusted to the change in plugin

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Rafał! :1st_place_medal:

File: src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php (hunk 2 above)
Comment (Rafikooo) [reply src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php:228]: https://github.com/Sylius/Sylius/commit/865d200fa334292df80ee558ef68aa30bf72dd66#diff-3b1a140e84b709ba909293ab73140c150f1fac8c177cc5485d64b866d2732850R226

File: src/Sylius/Bundle/AdminBundle/Resources/translations/messages.en.yml (hunk 4 above)
Comment (Rafikooo) [reply src/Sylius/Bundle/AdminBundle/Resources/translations/messages.en.yml:64]: https://github.com/Sylius/Sylius/commit/865d200fa334292df80ee558ef68aa30bf72dd66#diff-5b53765a5839c133bab43c07c01dfba2f2c345d3133e48a4492a31bdffe11cb3

**Pull request merged**